pkg/flow/internal/controller/loader.go

```diff
@@ -416,7 +416,7 @@ func (l *Loader) EvaluateDependencies(parentScope *vm.Scope, c *ComponentNode) {
 		return nil
 	})
 
-	if l.globals.OnExportsChange != nil {
+	if l.globals.OnExportsChange != nil && l.cache.HasModuleExportsChangedSinceLastCall() {
```

Comment:
This seems a little error prone since it updates the state of the cache. What do you think about having some kind of `ModuleExportsChangedTime()` method and comparing the changed time before/after the evaluate? 

Alternatively instead of returning the changed time, you can have it keep track of some kind of changed index and compare those instead.

Reply:
Instead of using a time, switched to using an incrementing index.